### DISCUSSION TITLE
Retrieve properties of component while testing

### DISCUSSION THREAD

**Discussion by Sch-Tim** [livewire/livewire#4770] [1. Help] [answered]
I made a component which consists of dynamically addable blocks. 
The blocks are stored in an associative array which keys are random strings. 

The problem is, that testing this is pretty hard. For further testing i need to know the randomly generated strings, so I need to access the array. 
While there is the `assertSet()` function to test a property, I couldn't find a function to retrieve a property. 
Is there such a feature? If not, it would be great to add it. 
Some function like `getProperty($name)`.

**Comment by joshhanley** [comment, marked as answer]
@Sch-Tim there is already a `->get($property);` method.

https://github.com/livewire/livewire/blob/f6d98ac775dd5e1b1a12dd3c2646e146216bc50c/src/Testing/TestableLivewire.php#L219

Can you try that and see if it works?

Hope this helps!

**Reply by Sch-Tim** [reply]
I'm sorry! I confused the get() method with the get(route) method and didn't even think about trying that to retrieve a property.

**Reply by joshhanley** [reply]
@Sch-Tim no worries! 🙂
